libr/bin/bin.c

```diff
@@ -519,9 +518,13 @@ R_API RBinPlugin *r_bin_get_binplugin_by_bytes(RBin *bin, const ut8 *bytes, ut64
 R_API RBinXtrPlugin *r_bin_get_xtrplugin_by_name(RBin *bin, const char *name) {
 	RBinXtrPlugin *xtr;
 	RListIter *it;
-	if (!bin || !name) {
+
+	r_return_val_if_fail (bin, NULL);
+
+	if (!name) {
```

Comment:
why not use returnvalifail with name too? imho its really confusing because this makes people not know when a function precondition is critical, just expected behaviour or something that shouldnt even be checked.

Reply:
that's exactly the thing. They are two different things. r_return_*if_fail means a precondition that the programmer should respect when calling the function. This check is here to handle a valid input of the function.